### DISCUSSION TITLE
Does prisma 2 support group by, and complex queries ?

### DISCUSSION THREAD

**Discussion by al-ani** [prisma/prisma#2095] [Q&A] [answered]
I have two concerns about Prisma 2 which are:

1- Prisma 1 does not support "Group By", so does Prisma 2 support now?

2- Does Prisma 2 support complex query?, like this below query.

      query {
      users(where: { id:"server1"  }) {
          id
          name
          lastMonthPaid  <-----------I want to use this variable in sub query
          offers{
            books (where:{date: lastMonthPaid}){<----------this is the sub query
              dealprice
            }
          }
        }
      }

**Comment by ryands17** [comment, marked as answer]
Hi @al-ani 

> 1- Prisma 1 does not support "Group By", so does Prisma 2 support now?

As for aggregations, you can track the issue [here](https://github.com/prisma/prisma-client-js/issues/5)

> 2- Does Prisma 2 support complex query?, like this below query.

Regarding sub-queries, you can follow this answer given [here](https://github.com/prisma/prisma/issues/2245#issuecomment-617038028) that shows how you can perform sub-queries via `select` and `include`.
We will be adding that to the docs soon and you can follow that issue [here](https://github.com/prisma/prisma/issues/2264)



Do let me know if this solves your query :)

**Reply by aymannav6** [reply]
thanks

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
